File: problems/1608-special-array-with-x-elements-greater-than-or-equal-x/solution.c
int specialArray(int* nums, int numsSize) {
    int count;
    for (int x = 0; x <= numsSize; x++) {
        count = 0;
        for (int i = 0; i < numsSize; i++) {
            if (nums[i] >= x) {
                count++;
            }
        }
        if (count == x) {
            return x;
        }
    }
    return -1;
}

File: problems/1608-special-array-with-x-elements-greater-than-or-equal-x/solution.java
import java.util.Arrays;

class Solution {
    public int specialArray(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length;
        for (int x = 1; x <= n; x++) {
            int count = 0;
            for (int i = 0; i < n; i++) {
                if (nums[i] >= x) {
                    count = n - i;
                    break;
                }
            }
            if (count == x) {
                return x;
            }
        }
        return -1;
    }
}